qiskit_experiments/curve_analysis/drawer_mixin.py

```diff
@@ -0,0 +1,512 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Curve drawing functionality for analysis."""
+
+import functools
+from typing import Callable, Set, List, Dict, Sequence, Union, Optional
+
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.ticker import FuncFormatter, ScalarFormatter
+from qiskit.utils import detach_prefix
+from uncertainties import unumpy as unp, UFloat
+
+from qiskit_experiments.curve_analysis.curve_data import FitData
+from qiskit_experiments.framework import AnalysisResultData, Options
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+
+
+class CurveDrawerMixin:
+    """Mixin to provide drawing functionalities for curve fit result."""
```

Comment:
Can you add a few more details regarding this mix-in? I.e. what drawing functionalities does it provide? When is it called in a curve analysis etc...

Reply:
We should also mention what sub-classes need to implement. As far as I can tell it looks like they need to implement:
```python
def _default_draw_options(cls):
```
correct?

Reply:
Yes, I'll update the docs after we finalize the design. I'm still wondering mixin vs instance (see my long reply above). You are correct, what the subclass may want to implement is `_default_draw_options`. But nothing stops you to override other methods (if you need fancy visualization).

Reply:
With this change https://github.com/Qiskit/qiskit-experiments/pull/738/commits/09a16bf779b119423d1feca7048533695a9de512 `_default_draw_options` has been removed. Now developer needs to directly update drawer instance option via analysis options.